CAToolsetPerfDataCollector/PerfDataCollector.cpp

```diff
@@ -0,0 +1,306 @@
+#include "PerfDataCollector.h"
+#include <algorithm>
+
+AnalysisControl PerfDataCollector::OnStopActivity(const EventStack& eventStack)
+{
+    //
+    // Overall C1, C1/analyze, C1 Passes
+    // 
+
+    // FE Pass
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetPassDuration<FrontEndPass, &PerfData::FEPass>);
+    // CA Pass
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetPassDuration<CodeAnalysisPass, &PerfData::CAPass>);
+    // BE Pass
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetPassDuration<BackEndPass, &PerfData::BEPass>);
+
+    //
+    // Activities in FE during CA Pass
+    //
+
+    // AST Creation
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<AstCreation, &PerfData::ASTCreation>);
+    // SST Clients, e.g., PREfast
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<CodeAnalysisPlugins, &PerfData::ASTClients>);
+
+    //
+    // Activities in PREfast during CA Pass
+    //
+
+    // CA PREfast's Function Analysis (including its own plug-ins, EspXEngine, and EspXEngine extensions)
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<CodeAnalysisFunction, &PerfData::CAFunction>);
+    // PREfast's FPA Function Analysis (FPA = Function Path Analysis)
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<CodeAnalysisPREfastFpaFunction, &PerfData::FPAFunction>);
+    // EspXEngine's CFG Building
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<EspXEngineCfgBuild, &PerfData::EspXCfgBuild>);
+    // EspXEngine's All Checks (BufferCheck and extensions, whichever are enabled)
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<EspXEngineChecks, &PerfData::EspXAllChecks>);
+    // EspXEngine's PathSensitiveChecks (BufferCheck and some of the extensions, whichever are enabled)
+    MatchEventStackInMemberFunction(eventStack, this,
+        &PerfDataCollector::GetCodeAnalysisEventDuration<EspXEnginePathSimulation, &PerfData::EspXPathSensitiveChecks>);
+
+    // Tell the analysis driver to proceed to the next event
+    return AnalysisControl::CONTINUE;
+}
+
+// Prints the per-TU performance data in CSV format. The first column is the file path.
+// The rest of the columns are the performance data in microseconds.
+void PerfDataCollector::PrintRecords()
+{
+    constexpr auto allZero = PerfData{};
+
+    // Print Header
+    std::wcout << L"File Path, " << PerfData::PerfDataHeader << std::endl;
+
+    for (const auto& [pathId, data] : perfDataPerTu)
+    {
+        // If CA Pass is less than FE Pass, it is for files that are not target of Code Analysis,
```

Comment:
I think we might want to write out Code analysis everywhere instead of using abbreviations.

Reply:
I will leave it as is in the code base, adding comment for the acronym(s) used. I will replace them in the UI texts.